python/langsmith/evaluation/_arunner.py

```diff
@@ -1021,6 +1046,46 @@ def _get_run(r: run_trees.RunTree) -> None:
         )
 
 
+def _include_attachments(
+    target: Union[ATARGET_T, Iterable[schemas.Run], AsyncIterable[dict]],
+) -> bool:
+    """Whether the target function accepts attachments."""
+    if _is_langchain_runnable(target) or not callable(target):
+        return False
+    # Check function signature
+    sig = inspect.signature(target)
+    params = list(sig.parameters.values())
+    positional_params = [
+        p for p in params if p.kind in (p.POSITIONAL_ONLY, p.POSITIONAL_OR_KEYWORD)
+    ]
+
+    if len(positional_params) == 0:
+        raise ValueError(
+            "Target function must accept at least one positional argument (inputs)"
+        )
+    elif len(positional_params) > 2:
+        raise ValueError(
+            "Target function must accept at most two positional "
+            "arguments (inputs, attachments)"
+        )
+    elif len(positional_params) == 2:
+        mismatches = []
+        for i, (p, expected) in enumerate(
+            zip(positional_params, ("inputs", "attachments"))
+        ):
+            if p.name != expected:
```

Comment:
wonder if we should only enforce this if p.default is not p.empty (ie if it doesn't have a default value). if it does have a default value then its technically fine that it has other positional args